### DISCUSSION TITLE
artisan route:cache memory exhaustion v3.0@beta

### DISCUSSION THREAD

**Discussion by derekcavaliero** [livewire/livewire#6000] [7. Bugs (archived)] [answered]
I noticed that after installing the v3 beta as instructed (https://livewire.laravel.com/docs/installation) that my deployments started failing due to `php artisan route:cache` hitting memory limits. I tried increasing it from 256MB to 512MB and even again to 1024MB and it always seems to exhaust memory. This happens both on our production server and on my localhost environment. When I run `php artisan route:cache` in my localhost environment with a 1024MB memory limit the console comes back with a `Segmentation fault: 11` message in the console.

I removed the `livewire/livewire` package and redeployed and the issue goes away. So it seems isolated to something inside the `livewire/livewire` v3 package.

Other environment details below, application is pretty simple it's only been in existence for about 2 weeks. Not doing anything fancy with routes - theres maybe 4 custom routes in the `web.php` routes file that mainly point to static views.

```
"require": {
    "php": "^8.1",
    "guzzlehttp/guzzle": "^7.2",
    "laravel/framework": "^10.10",
    "laravel/sanctum": "^3.2",
    "laravel/socialite": "^5.8",
    "laravel/tinker": "^2.8",
    "livewire/livewire": "^3.0@beta",
    "mews/purifier": "^3.4",
    "rickwest/laravel-wordpress-api": "^1.1",
    "rossjcooper/laravel-hubspot": "^6.0",
    "sentry/sentry-laravel": "^3.6",
    "socialiteproviders/microsoft": "^4.2"
},
"require-dev": {
    "barryvdh/laravel-debugbar": "^3.8",
    "fakerphp/faker": "^1.9.1",
    "laravel/breeze": "^1.21",
    "laravel/pint": "^1.0",
    "laravel/sail": "^1.18",
    "mockery/mockery": "^1.4.4",
    "nunomaduro/collision": "^7.0",
    "phpunit/phpunit": "^10.1",
    "spatie/laravel-ignition": "^2.0"
}
```

**Comment by gehrisandro** [comment]
I ran into the same issue today as well.

And this has already been mentioned here: https://github.com/livewire/livewire/pull/5910

I was able to find a dirty solution locally, but currently trying to fix it properly.

**Comment by joshhanley** [comment, marked as answer]
This has been fixed in PR #6008 and released in v3 beta.5.
